crates/router/src/core/refunds.rs

```diff
@@ -114,6 +114,77 @@ pub async fn refund_create_core(
     .map(services::ApplicationResponse::Json)
 }
 
+#[instrument(skip_all)]
+pub async fn auto_refund_core(
+    state: &AppState,
+    merchant_account: domain::MerchantAccount,
+    payment_attempt: storage::PaymentAttempt,
+    key_store: domain::MerchantKeyStore,
+    req: refunds::RefundRequest,
+) -> RouterResponse<refunds::RefundResponse> {
+    let db = &*state.store;
+
+    let (merchant_id, payment_intent, amount);
+
+    merchant_id = &merchant_account.merchant_id;
+
+    payment_intent = db
```

Comment:
Why do we need payment_intent , can auto refund be performed with attempt alone, for that particular attempt.

Reply:
No, we need `payment_intent` as well; otherwise we'll need to write a separate flow for this case.